opentracing-noop/src/main/java/io/opentracing/NoopTracer.java

```diff
@@ -15,19 +15,25 @@
 
 import io.opentracing.propagation.Format;
 
-public final class NoopTracer implements Tracer {
+public interface NoopTracer extends Tracer {
+//    static NoopTracer create() {
+//        return NoopTracerImpl.INSTANCE;
+//    }
+}
 
-    public final static NoopTracer INSTANCE = new NoopTracer();
+final class NoopTracerImpl implements NoopTracer {
+    final static NoopTracer INSTANCE = new NoopTracerImpl();
 
     @Override
     public SpanBuilder buildSpan(String operationName) {
-        return NoopSpanBuilder.INSTANCE;
+        return NoopSpanBuilderImpl.INSTANCE;
     }
 
     @Override
     public <C> void inject(SpanContext spanContext, Format<C> format, C carrier) {}
 
     @Override
-    public <C> SpanBuilder extract(Format<C> format, C carrier) { return NoopSpanBuilder.INSTANCE; }
+    public <C> SpanBuilder extract(Format<C> format, C carrier) { return NoopSpanBuilderImpl.INSTANCE; }
```

Comment:
The signature in the Tracer API is to return span context
```
<C> SpanContext extract(Format<C> format, C carrier);
```
What is the use case when `extract()` needs to return a builder?

The 3 most common uses cases for extract that I know of are
  1. extracting from a request that has no tracing context => return some form of "nothing", which can still be used in `childof()`, but will be ignored and a new trace is started
  1. extracting from a request that does have tracing context => return real span context that is used as a parent for the new span
  1. in data pipelines, extracting tracing context from some storage and using it as some Reference type when starting a new span

Neither of these cases requires the return value to be a builder.

Note that there is a special case for Zipkin-like tracers for use case 2 above, where the child of an RPC span is actually a new span with the same SpanID. However, that special case does not require a change in the API, it can be handled internally, [as done in Jaeger tracer](https://github.com/uber/jaeger-client-java/blob/master/jaeger-core/src/main/java/com/uber/jaeger/Tracer.java#L229).